autogpt_platform/backend/backend/server/integrations/router.py

```diff
@@ -69,6 +70,9 @@ def callback(
 ) -> CredentialsMetaResponse:
     logger.debug(f"Received OAuth callback for provider: {provider}")
     handler = _get_provider_oauth_handler(request, provider)
+    code_verifier = creds_manager.store._get_code_verifier(
+        user_id, provider, state_token
+    )
```

Comment:
1. Please do this after the state token is verified
2. The leading underscore suggests this method is only for internal use, but here it's called externally